File: Prokast.Server/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Prokast.Server.Entities
{
    public class Product
    {
        public int ID { get; set; }
        public required string Name { get; set; }
        public required string SKU { get; set; }
        public required string EAN { get; set; }
        public required string Description { get; set; }
        public DateTime AdditionDate { get; set; } = DateTime.Now;
        public DateTime ModificationDate { get; set; } = DateTime.Now;

        [JsonIgnore]
        public virtual List<AdditionalDescription> AdditionalDescriptions { get; set; }
        [JsonIgnore]
        public virtual List<AdditionalName> AdditionalNames { get; set; }
        [JsonIgnore]
        public virtual List<CustomParams> CustomParams { get; set; }
        [JsonIgnore]
        public virtual List<DictionaryParams> DictionaryParams { get; set; }
        [JsonIgnore]
        public virtual List<Photo> Photos { get; set; }

        public virtual PriceList PriceList { get; set; }

        public int ClientID { get; set; }
        public virtual Client Client { get; set; }

        //public int? StoredProductID { get; set; }

        public virtual StoredProduct? StoredProduct { get; set; }
        
        [JsonIgnore]
        public virtual List<OrderProduct>? OrderProducts { get; set; } = [];
    }
}

File: Prokast.Klient/src/models/Product.ts
import { AdditionalField } from "./AdditionalField";
import { CustomParam } from "./CustomParam";
import { DictionaryParam } from "./DictionaryParam";
import { Photo } from "./Photo";
import { PriceList } from "./PriceList";

export interface ProductModel {
  name: string;
  sku: string;
  ean: string;
  description: string;
  additionalDescriptions: AdditionalField[];
  additionalNames: AdditionalField[];
  dictionaryParams: DictionaryParam[];
  customParams: CustomParam[];
  photos: Photo[];
  priceList: PriceList;
}

export interface ProductResponse {
  model: ProductModel;
  id: number;
  clientID: number;
  createdDate: string;
}